examples/ristretto/src/ristretto.rs

```diff
@@ -266,33 +305,73 @@ pub fn encode(u: RistrettoPoint) -> RistrettoPointEncoded {
     RistrettoPointEncoded::new().update_start(&s.to_byte_seq_le())
 }
 
+/// Decodes the given point in accordance with the IETF standard.
+/// Note: There are many byte-strings resulting in incorrect decodings.
+/// These all checked for, in accordance with the IETF standard.
+pub fn decode(u: RistrettoPointEncoded) -> DecodeResult {
```

Comment:
I believe this could be written in a more functional way, but this is a faithful representation of the standard.